solution-base/mongodb/Makefile

```
@@ -19,5 +18,21 @@ fetch-mongodb-sharded:
 		--untardir ${CHART_DIR}
 
 patch:
-	@git apply --check ${PATCH_FILES}
-	@git apply ${PATCH_FILES}
+	@if [ -z "${PATCH_FILES}" ]; then \
+		echo "No patch files found in ${PATCH_DIR}"; \
+		exit 1; \
+	fi
+	@echo "Checking patches..."
+	@for patch in ${PATCH_FILES}; do \
+		echo "Checking $$patch"; \
+		git apply --check "$$patch" || { echo "Patch check failed for $$patch"; exit 1; }; \
+	done
+	@echo "Applying patches..."
+	@for patch in ${PATCH_FILES}; do \
+		echo "Applying $$patch"; \
+		git apply "$$patch" || { echo "Patch apply failed for $$patch"; exit 1; }; \
```

not sure what is the benefit of "catching" the error: you already add a log indicating we are applying the patch, and `git apply` will fail with an error when the patch cannot be applied...
```suggestion
		git apply "$$patch" \
```